.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,24 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04",
+  "hostRequirements": {
+    "cpus": 2
+  },
+  "waitFor": "onCreateCommand",
+  "updateContentCommand": {
+    "Installing apt packages": "sudo apt-get -yqq update && sudo apt-get install -yqq python3-pip python3-xrootd"
+  },
+  "postCreateCommand": {
+    "Installing python packages": "python3 -m pip install -r requirements.txt"
```

Comment:
why is this different from "updateContentCommand"?

Reply:
See lifecycle of devcontainers, https://github.com/devcontainers/spec/blob/main/docs/specs/devcontainerjson-reference.md#lifecycle-scripts. Although they are triggered by different conditions, I'm using them here strictly because the different commands are executed one after another, and because this (theoretically) allows someone to update requirements.txt and restart their container without doing a rebuild. Also, postCreateCommand runs in background, while the apt-get install is running in blocking mode.

All this can be tweaked, I'm sure.

Reply:
Nice!